src/rail/creation/degradation/observing_condition_degrader.py

```diff
@@ -324,16 +324,45 @@ def get_pixel_conditions(self, pixel: int) -> dict:
     def assign_pixels(self, catalog: pd.DataFrame) -> pd.DataFrame:
         """
         assign the pixels to the input catalog
+        check if catalogue contains position information;
+        if so, assign according to ra, dec;
+        else, assign randomly.
         """
         pixels = self.maps["pixels"]
         if "weight" in list((self.maps).keys()):
             weights = self.maps["weight"]
             weights = weights / sum(weights)
         else:
             weights = None
-        assigned_pix = self.rng.choice(pixels, size=len(catalog), replace=True, p=weights)
+
+        if "renameDict" in self.maps.keys() and set(['ra','dec']).issubset(list(self.maps["renameDict"].keys())):
+            # if catalog contains ra, dec, but needs renaming
+            rakey=self.maps["renameDict"]['ra']
+            deckey=self.maps["renameDict"]['dec']
+            assigned_pix=hp.ang2pix(self.config["nside"], catalog[rakey].to_numpy(), catalog[deckey].to_numpy(), lonlat=True)            
+        elif set(['ra','dec']).issubset(catalog.columns):
+            # if catalog contains ra, dec, and no renaming
+            assigned_pix=hp.ang2pix(self.config["nside"], catalog['ra'].to_numpy(), catalog['dec'].to_numpy(), lonlat=True)
+        else:
+            # if catalog doesn't contain position information 
+            print("No ra, dec found in catalogue, randomly assign pixels with weights.")
+            assigned_pix = self.rng.choice(pixels, size=len(catalog), replace=True, p=weights)
+            
+            # in this case, also attach the ra, dec columns in the data:
+            ra, dec=hp.pix2ang(self.config["nside"],assigned_pix,lonlat=True)
+            skycoord = pd.DataFrame(np.c_[ra,dec], columns=["ra","decl"])
+            catalog = pd.concat([catalog, skycoord], axis=1)
+            
+        # this is the case where there are objects outside the footprint\
+        if not (np.in1d(assigned_pix, pixels)==True).all():
```

Comment:
Nit: Depending on how large the `assigned_pix` array is, this could be somewhat inefficient. If you convert `assigned_pix` to a set, and then use `np.in1d(assigned_pix_set, pixels, assume_unique=True)` . 

But if you anticipate `assigned_pix` being relatively small though, then it's probably not worth the refactor.

Reply:
Now implemented.